tokio-trace/tokio-trace-core/src/callsite.rs

```diff
@@ -24,6 +24,33 @@ struct Registry {
     dispatchers: Vec<dispatcher::Registrar>,
 }
 
+type DispatchersChanged = bool;
+
+impl Registry {
+    fn establish_interest(&self, callsite: &'static Callsite) -> DispatchersChanged {
+        let mut any_dropped = false;
+        let meta = callsite.metadata();
+
+        self.dispatchers
+            .iter()
+            .for_each(|registrar| match registrar.try_register(meta) {
+                Some(interest) => callsite.add_interest(interest),
+                None => any_dropped = true,
+            });
+
+        any_dropped
+    }
+
+    fn reestablish(&mut self) {
```

Comment:
Also, I'm not a _huge_ fan of the "establish"/"reestablish" naming here --- it seems kind of wordy but also not all that descriptive. Not a huge deal, as these are internal functions and not APIs, but I thought I'd mention it.

Reply:
yea I suck at naming. I didn't know what to call it so this is what I landed on. I'm fine with naming it anything you want.

Reply:
I can't immediately think of something better, but I'll keep thinking on it.